lib/spack/spack/cmd/logs.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import gzip
+import os
+import shutil
+import sys
+
+import spack.cmd
+import spack.util.compression as compression
+from spack.cmd.common import arguments
+
+description = "print out logs for packages"
+section = "basic"
+level = "long"
+
+
+def setup_parser(subparser):
+    arguments.add_common_arguments(subparser, ["spec"])
+
+
+def _dump_byte_stream_to_stdout(instream):
+    if sys.platform == "win32":
```

Comment:
Instead of branch on a platform, just attempt to do one of these things and if it OSErrors fall back to something else.

Reply:
For now this is updated to use `.fileno()` everywhere. 

It seems to work on Windows: I'm less familiar with Windows IO, and given that https://docs.python.org/3/library/sys.html#sys.stdout specifically mentioned using `stdout.buffer`, I thought I'd do it the way they recommend vs. looking up whether a `fileno` would always be available (and whether `fdopen` would always be a reasonable way to interact with it).

https://docs.python.org/3/library/sys.html#sys.stdout does say though that `sys.stdout` is a [file object](https://docs.python.org/3/glossary.html#term-file-object), and on Windows it should have a file handle in all the contexts where it makes sense to invoke `spack logs`.